update to beta 0.2.7

ourownstory opened the pull request.

ourownstory committed:
update nb

ourownstory committed:
fix train-speed issue and add tests

ourownstory committed:
reset debug

ourownstory committed:
Merge pull request #170 from ourownstory/fix-train-speed

fix train-speed issue and add tests

ourownstory committed:
Merge pull request #165 from ourownstory/season-air

Multiplicative Seasonality modelling example: air passengers

ourownstory committed:
Merge branch 'master' into sub-daily-yosemite

ourownstory committed:
improve notebook

ourownstory committed:
add trend flex

ourownstory committed:
add narration

ourownstory committed:
add AR

ourownstory committed:
add comments

ourownstory committed:
fix missing lambda_delay init

ourownstory committed:
Merge pull request #172 from ourownstory/fix-lambda-delay

fix missing lambda_delay init

ourownstory committed:
Merge branch 'master' into sub-daily-yosemite

ourownstory committed:
finalize nb

ourownstory committed:
Merge pull request #166 from ourownstory/sub-daily-yosemite

Sub daily data example: yosemite temps

ourownstory committed:
add ar nb

ourownstory committed:
fix lower bound batch, epochs when applying train_speed

ourownstory committed:
Merge branch 'master' of github.com:ourownstory/neural_prophet

ourownstory committed:
Merge branch 'master' into ar-nb

ourownstory committed:
narrate notebook

ourownstory committed:
finnish write-up

ourownstory committed:
shorten nb

ourownstory committed:
update nb

ourownstory committed:
Merge pull request #173 from ourownstory/ar-nb

example_notebooks: Autoregression on Yosemite temps

ourownstory committed:
bump to 0.2.7

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

pep8speaks commented:
Hello @ourownstory! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`neuralprophet/__init__.py`](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/__init__.py):

> [Line 11:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/__init__.py#L11): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 25:1](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/__init__.py#L25): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file

* In the file [`neuralprophet/configure.py`](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py):

> [Line 14:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L14): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 41:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L41): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 42:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L42): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 79 characters)
> [Line 64:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L64): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 125:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L125): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 79 characters)
> [Line 129:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L129): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 155:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L155): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 157:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L157): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 160:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L160): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 167:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L167): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 174:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L174): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 79 characters)
> [Line 179:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/configure.py#L179): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)

* In the file [`neuralprophet/df_utils.py`](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/df_utils.py):

> [Line 329:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/df_utils.py#L329): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 332:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/df_utils.py#L332): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)

* In the file [`neuralprophet/forecaster.py`](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py):

> [Line 70:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L70): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 100:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L100): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 101:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L101): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 102:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L102): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 79 characters)
> [Line 105:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L105): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 106:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L106): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 79 characters)
> [Line 109:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L109): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 113:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L113): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 116:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L116): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 79 characters)
> [Line 117:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L117): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (116 > 79 characters)
> [Line 118:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L118): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 119:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L119): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (113 > 79 characters)
> [Line 258:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L258): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 79 characters)
> [Line 278:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L278): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (120 > 79 characters)
> [Line 302:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L302): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 311:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L311): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 312:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L312): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 79 characters)
> [Line 321:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L321): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 79 characters)
> [Line 366:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L366): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 79 characters)
> [Line 371:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L371): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (119 > 79 characters)
> [Line 372:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L372): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 375:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L375): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 376:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L376): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (95 > 79 characters)
> [Line 395:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L395): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 444:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L444): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 485:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L485): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 610:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L610): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (116 > 79 characters)
> [Line 636:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L636): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 647:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L647): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (119 > 79 characters)
> [Line 703:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L703): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 705:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L705): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 79 characters)
> [Line 729:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L729): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 747:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L747): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (120 > 79 characters)
> [Line 770:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L770): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 79 characters)
> [Line 781:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L781): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 794:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L794): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 798:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L798): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 831:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L831): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (99 > 79 characters)
> [Line 833:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L833): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 838:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L838): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 1122:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L1122): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (116 > 79 characters)
> [Line 1157:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/forecaster.py#L1157): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (119 > 79 characters)









* In the file [`neuralprophet/time_dataset.py`](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/time_dataset.py):

> [Line 128:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/time_dataset.py#L128): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)

* In the file [`neuralprophet/time_net.py`](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/time_net.py):

> [Line 90:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/time_net.py#L90): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 79 characters)
> [Line 137:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/time_net.py#L137): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 196:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/time_net.py#L196): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)

* In the file [`neuralprophet/utils.py`](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/utils.py):

> [Line 337:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/utils.py#L337): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 79 characters)
> [Line 339:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/utils.py#L339): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 465:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/utils.py#L465): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 484:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/utils.py#L484): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (95 > 79 characters)
> [Line 485:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/utils.py#L485): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 79 characters)
> [Line 489:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/neuralprophet/utils.py#L489): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)





* In the file [`tests/test_integration.py`](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py):

> [Line 49:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L49): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 73:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L73): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 116:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L116): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 130:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L130): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 139:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L139): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 160:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L160): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 347:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L347): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 366:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L366): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 372:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L372): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 441:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L441): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 450:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L450): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 459:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L459): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 477:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L477): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 491:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_integration.py#L491): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)

* In the file [`tests/test_unit.py`](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py):

> [Line 55:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L55): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 129:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L129): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (114 > 79 characters)
> [Line 132:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L132): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 146:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L146): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 147:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L147): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 159:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L159): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 184:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L184): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 79 characters)
> [Line 190:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L190): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 200:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L200): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 212:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L212): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (95 > 79 characters)
> [Line 216:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L216): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 220:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L220): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 225:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L225): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 229:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L229): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 233:80](https://github.com/arita37/neural_prophet/blob/f7e546b3b2ecf284f08dec8ed199229be0429331/tests/test_unit.py#L233): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)



##### Comment last updated at 2020-12-11 04:04:49 UTC

ourownstory committed:
Update README.md

ourownstory committed:
move set_log_level to utils, update example notebooks (#178)

    move to utils
    import logger in init
    update notebooks
    add 24-hour forecast in AR Notebook
    update debug_logger

ourownstory committed:
move set_log_level to utils, update example notebooks (#178)

    move to utils
    import logger in init
    update notebooks
    add 24-hour forecast in AR Notebook
    update debug_logger

ourownstory committed:
Update README.md

ourownstory committed:
upgrade torch requirements, capture warnings as logs (#194)

nn.ParameterDict will throw warning logs if used with torch==1.7.0. Will be resolved in future torch versions.

ourownstory committed:
Fix wrong missing data infill with non-daily data in split_df, remove all  data_freq defaults (#195)

* remove all freq defaults

* tests for df_split

* update docstring

* separate other data freq tests

ourownstory committed:
remove potential_datasets.md

ourownstory committed:
Add tests, Speedup tests, improve lr-range-test (#199)

ourownstory committed:
works on local python 3.6.9 (#200)

ourownstory committed:
Add colab support and prettify logger printout (#206)

* rename logger to NP
* change log format
* add open in colab link to all notebooks
* google colab notebooks support

ourownstory committed:
Fix handle missing (#207)

ourownstory committed:
add py version, git version and license badges

ourownstory committed:
Create .gitattributes

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
docstring discont trend

ourownstory committed:
Merge remote-tracking branch 'origin/master'

ourownstory committed:
add Crossvalidation util (#226)

* add crossvalidation
* fix set_log_level

ourownstory committed:
crossvalidation helper function in forecaster (#227)

ourownstory committed:
Create CNAME

ourownstory committed:
Create CNAME

ourownstory committed:
Update README.md

ourownstory committed:
bump version number

ourownstory committed:
Training hyperparameters improvement (#260)

* increase batch and epoch settings

* change to SGD

* softer Once-cycle

ourownstory committed:
Training improvements (#261)

* deeper cycle

* fix multiplicative components

* normalization: soft default (with zero as min)

* separate lr_test

* make optimizer hyperparameter

AirFin committed:
NeuralProphet_document_zh (#253)

NeuralProphet_document_zh

AirFin committed:
NeuralProphet_document_zh (#253)

NeuralProphet_document_zh

ourownstory committed:
Callable loss (#263)

* callable loss

* add test case

* documentation

Co-authored-by: farzinshams <farzinshams95@gmail.com>

ourownstory committed:
Robustify changepoints data format  (#264)

* make list of changepoints
* add documentation
* add tests, make more resilient

ourownstory committed:
require log_level in logger util (#265)

ourownstory committed:
Rename tqdm, remove overbleed option (#266)

* rename tqdm progress bar ; remove inputs_overbleed option

ourownstory committed:
Reg schedule: increasing regularization in last third of training (#268)

* migrate to config: get_scheduler, get_reg_weight
* add weight delay of regularization
* fine-tune ar-reg function
* create notebook ar-sparsity

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

HansikaPH committed:
bug fix in plot country holidays (#296)

* bug fix in plot country holidays

* fix plot lagged reg bug

Co-authored-by: ourownstory <ourownstory@users.noreply.github.com>

GonzagueHenri committed:
Update trend.md (#288)

tredn_smoothndess is derpecated and not in the current NP version

GonzagueHenri committed:
Update trend.md (#288)

tredn_smoothndess is derpecated and not in the current NP version

italoraony committed:
Update Dev Install command for README.md (#300)

Dev Install section is missing quotes around ".[dev]".

Tested on macOS.

profrodai committed:
updating the README with contributors

profrodai committed:
Including a reference about the Slack community in the README

ourownstory committed:
Create get_started_contributing.md

ourownstory committed:
Update get_started_contributing.md

ourownstory committed:
Create contributors.md

ourownstory committed:
Update README.md

ourownstory committed:
Create changelogs.md

ourownstory committed:
Update README.md

ourownstory committed:
Update get_started_contributing.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
disable log file by default (#348)

GonzagueHenri committed:
Energy datasets (#321)

Add 3 energy datasets including README.md to specify where the data is from (according to their license)

* 2 PV solar energy generation
* 1 energy consumption

ourownstory committed:
simplify test lag reg (#349)

ourownstory committed:
Rename get_started_contributing.md to contributing.md

ourownstory committed:
Update README.md

ourownstory committed:
Update contributing.md

ourownstory committed:
Update contributing.md

ourownstory committed:
Update contributing.md

ourownstory committed:
Update contributing.md

ourownstory committed:
Introduce issue templates

ourownstory committed:
Update bug_report.md

ourownstory committed:
Update bug_report.md

ourownstory committed:
Update bug_report.md

ourownstory committed:
Update bug_report.md

ourownstory committed:
Update feature_request.md

ourownstory committed:
Update bug_report.md

ourownstory committed:
Rename contributing.md to CONTRIBUTING.md

ourownstory committed:
Update README.md

ourownstory committed:
Create CODE_OF_CONDUCT.md (#352)

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update CONTRIBUTING.md

nlaptev committed:
Update README.md (#360)

ourownstory committed:
Chiragsamal add lagged regressors (#362)

* Added option to supply list of Lagged regressors 
* updated tests

Co-authored-by: Chirag Samal <chirag6891@gmail.com>

ourownstory committed:
Chiragsamal add lagged regressors (#362)

* Added option to supply list of Lagged regressors 
* updated tests

Co-authored-by: Chirag Samal <chirag6891@gmail.com>

ourownstory committed:
Improve train reg hyperparam (#374)

* increase batch and epoch settings

* Merge new configuration train code with slower training branch

* merge lr-rate-reduce2 with reg-improve2

* increase batch and epoch settings

* merge w old lr-range improvement

* skip softer cycle

* increase auto-epochs

* keep create_optimizer

ourownstory committed:
delete nb copies (#375)

ourownstory committed:
delete reg nb

ourownstory committed:
add double crossvalidation: once for validation and once for testing (#377)

* add double crossvalidation

* fix docstring

HansikaPH committed:
Sobyah fix issue 53 (#380)

* Fix Issue #53
Fixed Issue: Data: fail with specific error message when data contains duplicate date entries.

* Test Case for #53
Test case added for fix of Issue#53

* test case update

Co-authored-by: Sobia Iftikhar <Sobia@Sobias-MacBook-Pro.local>
Co-authored-by: Sobia Iftikhar <Sobia@dyn-118-139-78-100.its.monash.edu.au>

ourownstory committed:
Update contributors.md

ourownstory committed:
Delete contributors.md

ourownstory committed:
Update README.md

ourownstory committed:
Update df_utils.py

ourownstory committed:
Update forecaster.py

ourownstory committed:
Update df_utils.py

ourownstory committed:
Update forecaster.py

ourownstory committed:
Update forecaster.py

ourownstory committed:
fix missing y_shift in predict_trend (#390)

ourownstory committed:
Fix hdays (#398)

* upgrade minimum holidays package due to their issue 527

* add Indonesia to test

ourownstory committed:
Update holidays: Remove Holidays defined in python-holidays (#399)

* update holidays from prophet

* remove holidays in python-holidays

* convert warnings use to logger use

ourownstory committed:
Update changelogs.md

ourownstory committed:
Update CONTRIBUTING with  commands for black, unittests

ourownstory committed:
bump version number

ourownstory committed:
Global Modeling Update (#372) (#400)

* Global Modeling Update (#372)

* Global Modeling update

* Global Modeling revised according to comments

* Update test_global_modeling epochs and local_modeling to .fit()

* blackd

Co-authored-by: Mateus Gheorghe <mgheorghecr@gmail.com>

ourownstory committed:
remove manual debug all tests (#402)

ourownstory committed:
Remove attrdict and make all configs dataclasses (#403)

* removed all AttrDict

* convert holidays to dataclass

ourownstory committed:
Plot legend (#371) (#404)

* plot legend

Co-authored-by: Mateus Gheorghem <mgheorghecr@gmail.com>

StoneT2000 committed:
fix outdated link in energy data example (#363)

MateusGheorghe committed:
neural_prophet_dev_setup - error (#405)

Undo Windows git hook modification -> revert to linux

nishai committed:
Fix custom loss (#407)

* Allow Custom Loss function inheriting from _Loss (#406)
* Added test for custom loss extending _Loss

Co-authored-by: Nishai Kooverjee <30477593+nishai@users.noreply.github.com>
Co-authored-by: ourownstory <tabletop@DESKTOP-9ECQPIK.localdomain>

ourownstory committed:
Update README.md

ourownstory committed:
update Introduction Presentation

ourownstory committed:
expand documentation on yhat (#413)

kaleming committed:
Fix Symposium file path (#412)

Although the Symposium file name was updated, the README.md are still redirecting to the old path.

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Add ERCOT data (#415)

* add ercot data

* fix data format

* fixed minute display

ourownstory committed:
Ercot regressor example notebook (#416)

ourownstory committed:
Initial commit

ourownstory committed:
Create LICENSE

ourownstory committed:
Update README.md

ourownstory committed:
remove example notebooks (#422)

ourownstory committed:
Update README.md

ourownstory committed:
Migrate example data to neuralprophet-data repository (#423)

* add test-data

* fix data-dir in tests

* delete data

ourownstory committed:
Update pyproject.toml

ourownstory committed:
Merge branch 'docs'

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
remove docs (#424)

ourownstory committed:
Update README.md

ourownstory committed:
fix broken links

ourownstory committed:
Update development_timeline.md

ourownstory committed:
add neuralprophet as submodule

ourownstory committed:
Update README.md

ourownstory committed:
add neuralprophet-tutorials as submodule

ourownstory committed:
Merge branch 'tutorials'

ourownstory committed:
remove submodules

ourownstory committed:
add tutorials to docs

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
re-merge docs

ourownstory committed:
Merge pull request #431 from ourownstory/docs-return

Docs return

ourownstory committed:
Update README.md

ourownstory committed:
Update README.md

ourownstory committed:
Benchmark Framework (#438)

* add benchmark framework
* replace CV nb with benchmark nb
* add note about lack of lagged input support

ourownstory committed:
Benchmarking: improved interface (#439)

* benchmarking simplified

* replace previous benchmarking setup

* separate benchmarking and crossvalidation

ourownstory committed:
Rename single_ to _ (#445)

* rename _handle_missing_data to handle_missing_data

* rename single_ functions to start with just _

ourownstory committed:
Modularize predict and add predict_raw (#446)

* modularize predict and allow for raw predictions.

ourownstory committed:
add collect_predictions notebook (#447)

ourownstory committed:
fixed dict calls of dataclass (#451)

ourownstory committed:
Skip make_future_dataframe when predicting (in-sample) (#450)

* bypass make_future for normal predictions and combine predict with predict_raw
* skip nans at very end wehn filling missing data in certain cases
* maybe extend added
* add examples
* update tutorials
* fix holidays dict call of dataclass
* reduce reg intensity ar 10x
* reduce trend reg 10x
* update use without predict in readme

ourownstory committed:
make sure normalisation and fill missing are only called once ever (#453)

ourownstory committed:
adjust tests to optional make_future (#454)

ourownstory committed:
fix typing issues (#456)

ourownstory committed:
add .DS_Store to gitignore

bhausleitner committed:
doc migration from mkdocs to sphinx (#457)

* doc migration from neuralprophet docs

* Delete .DS_Store

* Delete .DS_Store

* Delete .DS_Store

* updated path in readme

* edit dataset paths of additional notebooks

* add script to build nblink files

* draft for overwriting of nblinks

* update script to create fresh nblink files and update notebook headline structures

Co-authored-by: Oskar Triebe <ourownstory@users.noreply.github.com>

bhausleitner committed:
Updated CONTRIBUTING.md to include section for writing documentation and added docs/readme.md (#460)

* updated contribution md for writing and building documentation, add docs readme

* correct typos and update requirements to include sphinx

* minor corrections in setup and sphinx requirements

ourownstory committed:
Benchmark framework: Upgrade to include any model and metrics (#458)

* improve inner workings benchmark

* include prophet 

* Add split method to abstract dataclass for NeuralProphetModel and ProphetModel

* n_lags correcti0n within train/val split function

* Add dynamic error calculation to compare NProphet and Prophet

* add new error functions and code refactoring

* restructure and fix benchmark metrics

* fix n_lag first forecasts None

* allow for benchmarking without prophet installed

* add manual Benchmarking

* re-write and re-run notebooks

Co-authored-by: bhausleitner <bernhard.hausleitner@tum.de>
Co-authored-by: berni <58265021+bhausleitner@users.noreply.github.com>

ourownstory committed:
Explicit Validation dataframe for validate_each_epoch (#462)

* update cv notebook

* make epoch-wise validation require df explicitly

*  improve plot_live_loss

* fix tests

* update quickstart documentation

* fix inconsistent check_dataframe calls

ourownstory committed:
Energy example notebooks update (#465)

* update solar example
* complete hospital load notebook

ourownstory committed:
repeat lr-range-test 3 times, log10 avg over steep lr (#466)

ourownstory committed:
update ercot dataset name

bhausleitner committed:
implmented csv logging functionality (#464)

* implmented csv logging functionality

* add dynamic naming and default params

* refactor csv storing pattern

* uncomment main

* improve logging

* logging dir variable

* handle optional fields

Co-authored-by: ourownstory <ourownstory@users.noreply.github.com>

bhausleitner committed:
Add multiprocessing of experiments and crossvalidation (#468)

* update multiprocessing

* improve signature

* fix nested pool with threadpool

Co-authored-by: ourownstory <ourownstory@users.noreply.github.com>

ourownstory committed:
Fix MASE to compute naive in-sample and add RMSSE (#469)

* fix mase error computation

* pass training truth to error functions

ourownstory committed:
make threads and processes optional (#470)

Co-authored-by: Table Top <tabletop@pop-os.localdomain>

ourownstory committed:
make threads and processes optional (#472)

ourownstory committed:
benchmark: optional multiprocessing and Verbose bench (#471)

* make threads and processes optional

* adding verbosity

ourownstory committed:
disable probress bar in bench

ourownstory committed:
Allow no metrics and Minimal training loop (#473)

* make minimal fitting procedure possible
* manual metrics

bhausleitner committed:
add n_forecasts handling to benchmark (#474)

ourownstory committed:
Multiprocessing improve reliability and CPU use (#475)

* garbage collect and no threadpool

* fix multiprocessing

Co-authored-by: dell desktop <delldesktop@pop-os.localdomain>

bhausleitner committed:
updated csv logging of summary (#476)

bhausleitner committed:
Update notes

ourownstory committed:
Add files via upload

ourownstory committed:
bump package version to 0.3.0

ourownstory committed:
Merge remote-tracking branch 'origin/master'

ourownstory committed:
Update setup.py

Commit:
update Facebook notes

bhausleitner committed:
revamp docs (#481)

bhausleitner committed:
update crossreferences and docs bootstrap grid (#482)

athatheo committed:
Fix NeuralProhet to NeuralProphet in documentation (#486)

bhausleitner committed:
Update README.md

ourownstory committed:
Update README.md

bhausleitner committed:
Pytest and Github Actions Build and Test (#490)

* resolve merge conflicts

* resolve merge conflicts

* Unittest to Pytest; Travis CI; Test Coverage. (#373)

* Create 新建文本文档.py

Test-commit

* pytest 1

* add travis ci

* test travis ci

* Update .travis.yml

* Update .travis.yml

* Update .travis.yml

* Update .travis.yml

* Update requirements.txt

* Update .travis.yml

* updating travis

* update travis

* updata travis

* Update .travis.yml

* Update requirements.txt

* Update requirements.txt

* Update requirements.txt

* Update .travis.yml

* Update .travis.yml

* Update .travis.yml

* coveragetest

* Update .travis.yml

* Update README.md

* 2 test

* Update .travis.yml

* coverage

* coverage

* Update .coverage

Co-authored-by: berni <58265021+bhausleitner@users.noreply.github.com>

* update test_integration and test_unit with pytest functionality

* update debug.py to entail all relevant integration and unit tests

* prevent coverage for site packages

* github action for workflow

* test the github action

* update workflow file

* update workflow

* update workflow to run on different os

* update badge

* update readme v2

* update tests

* edit workflows to only run on push and PR to master

* update pre push

* remove coverage

* update workflow yml

* update requirements and pytest trigger

* update yaml file

* update yaml

* edit yaml

* update ci

* update yml

* update

* update

* remove install of NP as dependency

* edit ci

* edit ci

* edit ci

* edit ci

Co-authored-by: santeFXneo <49849648+santeFXneo@users.noreply.github.com>
Co-authored-by: Oskar Triebe <ourownstory@users.noreply.github.com>

bhausleitner committed:
include codecov (#493)

* include codecov

* update workflow

* update codecov.yml to ignore tests and scripts

* update codecov README

* edit dev install requirements and developer docs for code coverage

ourownstory committed:
Update CONTRIBUTING.md

ourownstory committed:
Mateus df-copy fix (#494)


* Replaced .copy() in case of a list of dataframe

* list comprehension fixed

* simplifing make_list_dataframes

* fixing and optimizing global modeling code before dict implementation

* small fix on the split_df test for global modeling

* assert for data_params in local modeling and remove make_list_function

Co-authored-by: Mateus Gheorghe <mgheorghecr@gmail.com>

bhausleitner committed:
Update README.md

ourownstory committed:
Fix tutorials addition guide

yasirroni committed:
Update README.md (#500)

Add arXiv link for faster navigation

bhausleitner committed:
update nblink generation file (#503)

MateusGheorghe committed:
Freq problem v2 (#480)

* update code for automatically set freq of dataframe

* added check_test_freq and some warning for understanding freq detection

* update freq fixes with monthly exception

* update freq fixes with monthly exception

* solve yearly problem

* changes for accepting business day and business hour

* changes for accepting business day and business hour

* updated so it accepts quarterly frequencies

* remove debug.py

* add docstrings, delete make_list_dataframes, updated test_unit

Co-authored-by: Oskar Triebe <ourownstory@users.noreply.github.com>

bhausleitner committed:
Add cumulated downloads badge

bhausleitner committed:
Test for build pipeline error (#514)

* update readme

* update debug info

* update logging during pytest for ci

* update workflow

* update unit test

* add float typecast

* update new reg window

* update test_reg_delay settings

MateusGheorghe committed:
Updated test single test line (#518)

MateusGheorghe committed:
Small fix in split_df (#519)

MateusGheorghe committed:
Fix plot called as "PLOT" all caps

MateusGheorghe committed:
Solve test ubuntu issue (#528)

* Update neuralprophet_dev_setup

* test wsl

* add dtype to the np.array() inside _stride_lagged_features function

bhausleitner committed:
integration tests for additional international holidays (#529)

bhausleitner committed:
Incorporate integration tests for utility functions (#530)

* integration tests for additional international holidays

* add integration tests for dataframe utility functions

bhausleitner committed:
update test integration (#540)

ourownstory committed:
Local normalization for panel datasets (#541)

* local normalization
* change plot_model_parameters so it can work with local_modeling True
* local normalization only works with dict
* updated split_df for dicts
* add docstring to functions related to split_df
* add self._normalize function
* now the default for local modeling is global time data params
* standardize df-dict
* remove train_speed
Co-authored-by: Mateus Gheorghe <mgheorghecr@gmail.com>

ourownstory committed:
Move unknown_data_normalization to config_normalization (#542)

* move unknown-data-normalization to config_normalization

* update tests

* fix plot calls to unknown_data_normalization

ourownstory committed:
Create GlobalTimeDataset class from df_dict (#543)

ourownstory committed:
TimeDataset add metadata: df_name (#544)

* add metadata to dataset and loader with df_name field
* add loader test

ourownstory committed:
Introducing forgetting: Weighting loss based on recentness of observation. (#545)

* add newer sample weight param and remove loss reduction

* support loss reduction=none

* add checks

* add time-based weighing of loss

MateusGheorghe committed:
Global modeling example notebook (#499)

* add global modeling jupyter example  with ERCOT data

MateusGheorghe committed:
Removed unnecessary comments (#548)

ourownstory committed:
reduce defaults for auto epoch: 20-200 epochs (#547)

* shorten epochs, add test
* fix test newer-samples

MateusGheorghe committed:
Test global modeling events plus regressors (#557)

* Update neuralprophet_dev_setup

* test wsl

* added test_global_modeling_events_plus_regressors

MateusGheorghe committed:
Test global modeling plus events (#556)


* add test_global_modeling_plus_events

* comment plot_component functions

Co-authored-by: Oskar Triebe <ourownstory@users.noreply.github.com>

MateusGheorghe committed:
Test global modeling plus regressors (#555)


* added test_global_modeling_plus_regressors

* comment plot_component functions

Co-authored-by: Oskar Triebe <ourownstory@users.noreply.github.com>

epistoteles committed:
fix typo in autoregression_yosemite_temps.ipynb tutorial (#560)

ourownstory committed:
Expand tests for global modelling: readd events-regressors and separate lagged/future regressors (#559)

* re-add Test global modeling events plus regressors

* add future/lagged regressors test separately

ourownstory committed:
Fix make-future-dataframe bypassing dataframe (#562)

* add test for make-future

* fixed issue

epistoteles committed:
fix documentation parameter formatting (#561)

* add newlines before Args and Returns

* use consistent formatting for indicating parameter type

ourownstory committed:
Live plot nb (#563)

* start liveplotloss nb

* fix plot liveloss

* pass parameters

* pass parameters and update notebook

ourownstory committed:
Simplify progress info printing, speed up tests, remove epochs from fit (#564)

* start liveplotloss nb

* fix plot liveloss

* pass parameters

* pass parameters and update notebook

* streamline progress

* finish progress display

* fix typo

* collapse signature

* add test

* add LR to tests to speed up auto-lr

* fix bar undefined when exiting from plotting not ffound

MateusGheorghe committed:
Global modeling docstrings (#558)

* updated global modeling related functions doctrings

* changes in doctrings from plot_model_parameters

MateusGheorghe committed:
Infer freq raises warning even with none (#565)

* made freq=auto default

* add test for freq=None

Co-authored-by: Oskar Triebe <ourownstory@users.noreply.github.com>
